app/src/main/java/com/github/onlynotesswent/model/note/NoteRepositoryFirestore.kt

```diff
@@ -26,6 +26,47 @@ class NoteRepositoryFirestore(private val db: FirebaseFirestore) : NoteRepositor
       val image: String,
       val commentsList: List<String>
   )
+  /**
+   * Converts a single Comment object into a formatted string for Firestore storage.
+   *
+   * @param comment The Comment object to convert.
+   * @return A string representing the Comment, formatted as
+   *   "commentId<delimiter>userId<delimiter>userName<delimiter>content<delimiter>creationDate<delimiter>editedDate".
+   *
+   * Each field is separated by the `commentDelimiter` for easy parsing during retrieval.
+   */
+  private fun convertCommentToString(comment: Note.Comment): String {
+    return comment.commentId +
+        commentDelimiter +
+        comment.userId +
+        commentDelimiter +
+        comment.userName +
+        commentDelimiter +
+        comment.content +
+        commentDelimiter +
+        comment.creationDate.seconds.toString() +
+        commentDelimiter +
+        comment.editedDate.seconds.toString()
+  }
+  /**
+   * Converts a formatted string snapshot of a comment back into a Comment object.
+   *
+   * @param commentSnapshot The string representing the comment, formatted as
+   *   "commentId<delimiter>userId<delimiter>userName<delimiter>content<delimiter>creationDate<delimiter>editedDate".
+   * @return A Comment object created from the parsed string values.
+   * @throws IndexOutOfBoundsException if the comment snapshot is improperly formatted and does not
+   *   contain the expected number of fields.
+   */
+  private fun convertCommentStringToComment(commentSnapshot: String): Note.Comment {
+    val commentValues = commentSnapshot.split(commentDelimiter)
+    return Note.Comment(
+        commentValues[0],
+        userId = commentValues[1],
+        userName = commentValues[2],
+        content = commentValues[2],
```

Comment:
I believe this is an error, should be` content = commentValues[3],`, and adapt the following indexes accordingly